src/webhook_launcher/app/models.py

```diff
@@ -287,9 +288,15 @@ def revision(self):
 
     @property
     def lsr(self):
-        _lsr = self.lastseenrevision_set.all()
-        if _lsr:
-            return _lsr[0]
+        # TODO: refactor the WebHookMapping and LastSeenRevision relation
+        if not hasattr(self, '_lsr'):
+            if self.pk:
+                self._lsr, _ = LastSeenRevision.objects.get_or_create(
+                    mapping=self
+                )
+            else:
+                return None
+        return self._lsr
```

Comment:
It is something to keep in mind for my django console usage, where the WHM objects are quite long lived.

Fair enough.